src/utils/url.js

```diff
@@ -31,7 +31,13 @@ function _convert (v) {
   return _isBoolean(_isNull(_isNum(v)))
 }
 
-export function urlParams (url) {
+function _queryNameValue (query, predicate) {
+  const name = _decode(query[0])
+  const value = (isFunction(predicate) && predicate.length === 1 && predicate(name)) ? _decode(query[1]) : query[1]
```

Comment:
I would not do the `predicate.length === 1` check or at least change it to be `predicate.length >= 1` as it's fine to pass a function that expects more arguments in js world

Reply:
makes sense, we won't use the predicate, decided to go with getQueryParameter suggestion